src/tesseract.pybind.h

```diff
@@ -83,28 +84,116 @@ void add_tesseract_module(py::module& root) {
           "get_observables_from_errors",
           [](TesseractDecoder& self, const std::vector<size_t>& predicted_errors) {
             std::vector<bool> result(self.num_observables, false);
-            const auto& indices = self.get_flipped_observables(predicted_errors);
-            for (int index : indices) {
-              result[index] = true;
+            for (size_t ei : predicted_errors) {
+              for (int obs_index : self.errors[ei].symptom.observables) {
+                result[obs_index] = result[obs_index] ^ true;
+              }
             }
             return result;
           },
           py::arg("predicted_errors"))
       .def("cost_from_errors", &TesseractDecoder::cost_from_errors, py::arg("predicted_errors"))
       .def(
-          "decode",
-          [](TesseractDecoder& self, const std::vector<uint64_t>& detections) {
-            std::vector<bool> result(self.num_observables, false);
+          "decode_from_detection_events",
+          [](TesseractDecoder& self, const py::array_t<bool>& syndrome) {
```

Comment:
As mentioned above for simplex, this method (the one that takes detector measurements as input) should be called `decode`, and there should be a new method `decode_from_detection_events` that has the original function signature (i.e. it takes detection events as input, not the bitstring). See my above comment on simplex for more details.

```suggestion
          "decode",
          [](TesseractDecoder& self, const py::array_t<bool>& syndrome) {
```

Reply:
@oscarhiggott Thank you. Done.